src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -155,6 +155,40 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host.");
+
+            if (Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager))
+            {
+                _logger.LogDebug($"Host status: {scriptHostManager.State.ToString()}. Drain mode enabled: {drainModeManager.IsDrainModeEnabled}");
+
+                if (scriptHostManager.State != ScriptHostState.Running)
+                {
+                    _logger.LogDebug("Host is not running, cannot resume");
+                    return StatusCode(StatusCodes.Status409Conflict);
+                }
+
+                if (!drainModeManager.IsDrainModeEnabled)
+                {
+                    _logger.LogWarning("Resume request received when not in drain mode");
+                    return Ok(new { hostStatus = scriptHostManager.State.ToString() });
```

Comment:
Question: Is 200 OK the appropriate response code in this case? Should it be 405?

Reply:
Yeah we discussed this as part of the design review. If the host is running and is not in drain mode, we want to return 200 as that signals to Antares that they can use this host and it's ready for them. Anything other than a 200 could potentially lead to this worker being killed on their side and that would be a waste when it's running as expected

Reply:
If we need to return a body that the caller will be parsing, we should define and use a concrete Type

Reply:
Good point, I'll do that!

Reply:
Thoughts on just returning the existing HostStatus type?

```csharp
var status = new HostStatus { State = scriptHostManager.State.ToString() };
return Ok(status);
```

Reply:
Recommend defining a new ResumeStatus Type with a single HostState member for now

Reply:
I think that would change the data contract we have with the antares team from `{ "state": "running" }` to something like `{ "hostStatus": { "state": "running" } }`